.github/workflows/publish.yml

```
@@ -6,33 +6,73 @@
 # separate terms of service, privacy policy, and support
 # documentation.
 
-name: Upload Python Package
+name: Python Package Workflow
 
 on:
+  push:
+    branches:
+      - main
   release:
     types: [published]
+  schedule:
+    - cron: "0 17 * * FRI"  # Every Friday at 5 PM UTC
 
 permissions:
-  contents: read
+  contents: write
 
 jobs:
-  deploy:
+  pre_commit_and_tests:
+    if: github.event_name == 'push' && github.ref_name == 'main'
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
```

```suggestion
        uses: actions/setup-python@v5
```
See line 64 or let `uv` install Python.

---

I have updated it in latest commit https://github.com/achatap/browser-use/commit/bf9aca838bcb031785b9a1fae4e24127efb3886c